v4/unit_cache.go

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2022 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package work
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/uber-go/tally"
+)
+
+// UnitCache represents the cache that the work unit manipulates as a result
+// of entity registration.
+type UnitCache struct {
+	m sync.Map
```

Comment:
sync.Map was added in 1.19, will go version be changed in go.mod?

Reply:
i believe `sync.Map` was actually introduced in `1.9` (according to the documentation):

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/5921929/195494581-7cff153d-e086-4969-8a54-400e451454e0.png">

it's also worth calling out that the continuous integration should catch compilation errors - it builds the project against several versions of the runtime to ensure it works as expected!